directed_work/03_deploy_api/README.md

```diff
@@ -0,0 +1,318 @@
+![New project](img/2.png)
+
+### 2.2 Install the Google Cloud SDK
+
+1. Go to the [Google Cloud SDK](https://cloud.google.com/sdk/docs/install) page and follow the instructions to install the SDK for your operating system.
+
+#### 2.2.1 for Mac OS
+1. Download the `google-cloud-sdk` tar file from the [Google Cloud SDK](https://cloud.google.com/sdk/docs/install) page
+2. Extract the tar file in your home directory (just double click on the tar file)
+3. Open a terminal, you should be in your home directory
+4. Run `./google-cloud-sdk/install.sh` to install the SDK. You will be prompted to answer a few questions. You can just press `Enter` to accept the default answers or `y` to accept the default answers and start the installation.
+
+### 2.3 Initialize the Google Cloud SDK
+
+1. In your GUI (VScode, Pycharm, etc) or in your terminal, run `gcloud init` to initialize the SDK. This should appear in your terminal:
+
+```bash
+Welcome! This command will take you through the configuration of gcloud.
+
+Your current configuration has been set to: [default]
+
+You can skip diagnostics next time by using the following flag:
+  gcloud init --skip-diagnostics
+
+Network diagnostic detects and fixes local network connection issues.
+Checking network connection...done.                                                                                                                                                                                      
+Reachability Check passed.
+Network diagnostic passed (1/1 checks passed).
+
+You must log in to continue. Would you like to log in (Y/n)?
+```
+
+2. You will be prompted to log in to your GCP account. The following message should appear in your terminal:
+
+```bash
+You are logged in as: [google email address].
+
+Pick cloud project to use: 
+ [1] your-newly-created-project
+ [2] Create a new project
+Please enter numeric choice or text value (must exactly match list item):
+```
+3. Select the project you just created.
+
+### 2.4 Change the project
+If you want to work in another project, you can see the list of your projects by running the following command in your terminal:
+
+```bash
+> gcloud projects list
+```
+
+This should give you something like this:
+
+```bash
+PROJECT_ID                NAME                  PROJECT_NUMBER
+something-staging-2587    something-staging     804012817122
+something-production-24   something-production  392181605736
+```
+
+Now if you want to change the project you are working on, you can change the project by running the following command in your terminal:
+
+```bash
+> gcloud config set project something-staging-2587
+```
+
+where `something-staging-2587` is the `PROJECT_ID` of the project you want to work on.
+
+## 3. Deploy the app to GCP
+
+[OLD COMMAND ?]
+
+If you run now the following command in your terminal:
+
+```bash
+> gcloud run deploy flask-app --region=europe-west1 --source=$(pwd) --allow-unauthenticated
```

Comment:
I think (without researching so might be wrong) that GCP now allows to do both build and deploy in one command.
But I would still show the old way (build then deploy) and explain what's happening (build your image locally and then deploy it).
It's still very much what's happening behind the hood but GCP just abstracted one command away